New package: AutoAD v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: AutoAD
- Repository: https://github.com/EnzymeAD/AutoAD.jl
- Created by: @wsmoses
- Version: v0.1.0
- Commit: 79ab8e5a81b8b14448515805feb9f4549dc362e5
- Reviewed by: @wsmoses
- Reference: https://github.com/EnzymeAD/AutoAD.jl/commit/79ab8e5a81b8b14448515805feb9f4549dc362e5#commitcomment-164059497
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd60c3c73de98068164a69817d6bb86f1da734855fc22a24ffb92ca13d51eacf717a049e0afa5184bb3b7c773c89e9d232127cfb1cba1b68532261037588f5a46a824dbb2c86a53ccea116e292c7f99781433cf628f34229bc1060b3691c82dfe235b784710fccc636d36fe0f9c96f54d11c074a51b5007e4f79430332e1ed4f3480e26cee2dcf250757af55bc4696bfb39991eb126a741790d716f9bacc52d69790e037badcd563f886c4d0a0a3c2e0dd5 -->

ChrisRackauckas commented:
I think the bigger thing is, we were going to register a package with this name to JuliaDiff probably over the weekend. You can see the plan for it here: https://github.com/JuliaDiff/DifferentiationInterface.jl/issues/820#issuecomment-3078794878 . So I don't know about this name/domain being taken by EnzymeAD. While I would make this match the SciMLSensitivity auto-choice algorithm which does have an explicit preference for Enzyme (https://github.com/SciML/SciMLSensitivity.jl/blob/master/src/concrete_solve.jl#L9-L77), it has fallbacks to ReverseDiff, Zygote, and FiniteDiff as a way to help make sure it finds something that works.

I would like to work with folks like @wsmoses on this one, as you can see from that other very long thread that we have been in need of this for awhile. But I think that with this open of a name, it should really be dedicated to something like JuliaDiff that is a guiding maintainer organization for all AD usage rather than being tied to a specific set of maintainers because if it is done right, I think it needs to be a benchmark-based data-driven collaboration with Enzyme, Mooncake, ForwardDiff, ReverseDiff, etc. maintainers and not directly tied to one.

[noblock]

JuliaRegistrator committed:
New package: AutoAD v0.1.0

UUID: 3adc3437-ee8d-4879-8523-41c5a6d491e3
Repo: https://github.com/EnzymeAD/AutoAD.jl.git
Tree: f3f0ceffa45f0bbc9fbc453c6fe4f991fd31af46

Registrator tree SHA: 50f504d641745716a5b3eabaf681d3a4937d2ae3

goerz commented:
Yeah, this isn't ready yet for registration by the standards we normally apply. At a minimum, it would have to have some useful functionality and enough documentation that a user can get started with using the package.

The _name similarity_ as such is clearly a false positive, so I'm just going to override that. As for the larger discussion about how this package fits into the ecosystem and whether it should be managed as part of Enzyme or within SciML, I have no absolutely no horse in that race. But with a name this general, I would say there should be a clear consensus among the people involved in this part of the ecosystem. I'm not sure that's the case here yet, but this probably depends mainly on @ChrisRackauckas (who would also be in a position to merge this manually)

wsmoses commented:
Yeah that's fair (and apologies for my ignorance on the order of operations for registering / building, which I should've remembered when registering Reactant last year with a similar level of bare-bones ness: https://github.com/JuliaRegistries/General/pull/106843 ).

That said, taking a step back to make sure we proceed in a manner the community would be behind, does the current plan for building the package sound reasonable to folks?

Specifically, that being building a package which can take various autodiff settings, and device offloading capabilities, benchmarks these for a given function, and returns a thunk or other object which specifies what was found to be the most effective. Probably will use DI for autodiff codegen. And will add any/everyone to the repo interested in helping.

If folks are behind this plan (and other parts of the package seem fine to folks), I can go ahead and make an extremely minimal (and subject to change for all API parts/etc) implementation, and then re-ping the thread once we have a good placeholder to build off of?
